dapp/src/components/buySell/ApproveSwap.js

```diff
@@ -23,12 +23,13 @@ const ApproveSwap = ({
   rpcProvider,
   isMobile,
 }) => {
-  const [coinApproved, setCoinApproved] = useState(false)
   const [stage, setStage] = useState('approve')
   const [contract, setContract] = useState(null)
   const [isApproving, setIsApproving] = useState({})
   const web3react = useWeb3React()
   const { library, account } = web3react
+  const coinApproved = stage === 'done'
```

Comment:
No need for a separate state since coinApproved can be derived from `stage` state